Source/WebCore/rendering/svg/LegacyRenderSVGRoot.cpp

```diff
@@ -212,6 +212,7 @@ void LegacyRenderSVGRoot::layout()
 
     repainter.repaintAfterLayout();
 
+    setNeedsLayoutIfNeededAfterIntrinsicSizeChange();
```

Comment:
Calling setNeedsLayout.. from layout() is a hack. You only ever need to do something is response to view box change, so the place that has to do this is svgAttributeChanged() in SVGSVGElement.

Reply:
Let me explore. :-)